### PR TITLE
FIneract-1348-balloon-payment-fixed-principal-percentage

### PR DESCRIPTION
**Pull request by fynmanoj**
## Description
https://issues.apache.org/jira/browse/FINERACT-1348
Describe the changes made and why they were made.

Ignore if these details are present on the associated [Apache Fineract JIRA ticket](https://github.com/apache/fineract/pull/1284).

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [x] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [x] Create/update unit or integration tests for verifying the changes made.

- [x] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [x] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/api-docs/apiLive.htm with details of any API changes

- [x] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

**Comment by awasum** [comment]
Hi @fynmanoj  Sorry for the delay. There are now merge conflicts on the PR. 

@francisguchie , @bharathc27 you can test this function together with the UI changes found here: https://github.com/openMF/community-app/pull/3381

**Comment by fynmanoj** [comment]
@awasum , I have rebased the PR

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/Loan.java [hunk 1]
```diff
@@ -1622,6 +1627,13 @@ public void updateLoanSummarAndStatus() {
             this.fixedEmiAmount = null;
         }
 
+        if (command.isChangeInBigDecimalParameterNamed(LoanApiConstants.fixedPrincipalPercentagePerInstallmentParamName,
+                this.fixedPrincipalPercentagePerInstallment)) {
+            this.fixedPrincipalPercentagePerInstallment = command
+                    .bigDecimalValueOfParameterNamed(LoanApiConstants.fixedPrincipalPercentagePerInstallmentParamName);
+            actualChanges.put(LoanApiConstants.fixedPrincipalPercentagePerInstallmentParamName, this.fixedEmiAmount);
+        }
```
Comment (ptuomola) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/Loan.java:1651]: Shouldn't this be this.fixedPrincipalPercentagePerInstallment, not this.fixedEmiAmount?

**Review by ptuomola — CHANGES_REQUESTED** [review]

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/Loan.java (hunk 1 above)
Comment (avikganguly01) [reply fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/Loan.java:1651]: @fynmanoj : Can you confirm this and rebase it again? Rest of it looks good.

Comment (fynmanoj) [reply fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/Loan.java:1651]: done!

**Review by avikganguly01 — APPROVED** [review]

**Comment by awasum** [comment]
Please lets make it a habit not to merge PRs when Travis Tests have not passed. Can you close the corresponding issue on JIRA please...

**Pull request merged**